stdlib/binomial.hk

```diff
@@ -0,0 +1,31 @@
+# Hakaru Standard Library
+# author: Mahmoud Khattab
+
+#############################################
+# Binomial Distribution and Transformations #
+#############################################
+
+# Bernoulli distribution is simply Binomial Distribution of 1 experiment
+def bernoulli(p prob):
+	categorical([real2prob(1-p),p])
+
+# Binomial Distribution
+# Describes number of successes in n independant experiments
+# Each with success probability p
+def binomial(n nat, p prob):
+	arr <~ plate _ of n: bernoulli(p)
+	sum = summate i from 0 to n: arr[i]
+	return sum
+
+# Beta-Binomial is defined as a binomial distribution
+# with the success probability pulled from a beta distribution
+# Parameters a and b are the shape parameters of the beta distribution
+def betaBinomial(a prob, b prob, n nat):
+	p <~ beta(a,b)
+	binomial(n,p)
+# TODO: test betaBinomial correctness. 
+# i.e. betaBinomial(1,1,n) should be equivalent to rectangular(n)
+
+# Negative Hyper Geometric distribution
+def negativeHyperGeometric(n1 nat, n2 nat, n3 nat):
```

Comment:
Can you use better arguments names than `n1` `n2` `n3`? Also put a comment here stating the natural numbers `n3` is allowed to take on, since it is constrained?